ow_plexil/src/plexil-adapter/FaultDependencies.h

```diff
@@ -0,0 +1,65 @@
+#ifndef Fault_Dependencies
+#define Fault_Dependencies
+
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+// Fault Dependencies Class for use in the Interface class with fault lookups.
+
+struct Fault {
+  std::string name;
+  std::string parent_subsystem;
+  int faulty = 0;
+  std::vector<std::pair<std::string, std::string>> impacts;
+};
+
+struct Procedure {
+  std::string name;
+  int inoperable = 0;
+};
+
+struct Subsystem {
+  std::string name;
+  int local_fault = 0;
+  int non_local_fault = 0;
+  int faulty = 0;
+  int inoperable = 0;
+  std::vector<std::string> faults;
+  std::vector<std::pair<std::string, std::string>> impacts;
+};
+
+class FaultDependencies
+{
+public:
+
+  FaultDependencies (std::string file_name, bool verbose_flag);
+  ~FaultDependencies() = default;
+  FaultDependencies (const FaultDependencies&) = delete;
+  FaultDependencies& operator= (const FaultDependencies&) = delete;
+
+  // get and update functions
+  void updateFault(const std::string &name, int status);
+  std::vector<std::string> getActiveFaults(const std::string &name);
+  bool checkIsOperable(const std::string &name);
+  bool checkIsFaulty(const std::string &name);
+  void DebugPrint();
```

Comment:
State query and print functions can all be marked const
```suggestion
  std::vector<std::string> getActiveFaults(const std::string &name) const ;
  bool checkIsOperable(const std::string &name) const;
  bool checkIsFaulty(const std::string &name) const;
  void DebugPrint() const;
```

Reply:
Changed! It originally was giving me errors, turns out you need to use a special access function for maps to make them const compatible.

Reply:
Oh! Was it `at`? That makes sense. I believe the `[]` operator is kept in `std::map` merely for backwards compatibility, but is generally discouraged in favor of `at`. It's `const` overload is certainly part of the reason why.